pkg/kube/krt/fetch.go

```diff
@@ -30,8 +30,18 @@ func FetchOne[T any](ctx HandlerContext, c Collection[T], opts ...FetchOption) *
 	}
 }
 
+// FetchOrList runs a query against the provided collection and subscribes to updates if ctx is set.
+// If unset, this will just be a one time list operation.
+func FetchOrList[T any](ctx HandlerContext, cc Collection[T], opts ...FetchOption) []T {
```

Comment:
What's the intended usage?  Seems like a possible footgun

Reply:
The usage I have in mind is for checking if a Secret reference is allowed. We use this in two places:
* In the gateway controller to decide if we accept the Gateway
* In the XDS serving, to verify a gateway is actually allowed to access the certificate in the first place.
The former is in a KRT context so can use Fetch, while the latter is not.

Most things using krt are going to have some boundary where they interact with things outside krt. For example, ambient has the `All()` function that does a List. The `FetchOrList` is just useful when that same boundary is used by krt things and non-krt things

Reply:
Definitely possible to mis-use those (as with a plain list)